### ISSUE TITLE
Partially unknown type for __init__ for generic class?

### ISSUE THREAD

**Issue by Holt59** [microsoft/pyright#8046]
**Describe the bug**

When calling `__init__` manually (without relying on `super()`) on a generic class, a "partially unknown" error is raised.

**Code or Screenshots**

```python
class B[U]:
    def __init__(self): ...


class C[U](B[U]):
    def __init__(self):
        B.__init__(self)  # <-- error here
```

I know that in this case I could use `super()`, but the true code has multiple inheritance so it is hard to use `super()`. 

If this is expected but there is a way to change the code to not have the warning, I'll be happy to know it.

**VS Code extension or command-line**

Tested with both, pyright CLI is 1.1.365

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

Evaluating constructor calls in Python is very complex. You can get a sense for the challenges by reading the [typing spec chapter on constructors](https://typing.readthedocs.io/en/latest/spec/constructors.html).

If you want to avoid an `Unknown` in this case, you will need to call `B[U].__init__` rather than `B.__init__`.
